app/src/main/java/com/codepath/apps/critterfinder/services/LocationService.java

```diff
@@ -0,0 +1,115 @@
+package com.codepath.apps.critterfinder.services;
+
+import android.content.Context;
+import android.location.Address;
+import android.location.Geocoder;
+import android.location.Location;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.util.Log;
+
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.google.android.gms.location.LocationServices;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * Created by smacgregor on 2/29/16.
+ */
+public class LocationService implements GoogleApiClient.ConnectionCallbacks,
```

Comment:
A location service responsible for using the google play api for extracting the gps coordinates and then setting up an async task that will geo decode that lat/long into a postal code.